libmachine/provision/provisiontest/sshcommander.go

```diff
@@ -0,0 +1,22 @@
+//Package provisiontest provides utilities for testing provisioners
+package provisiontest
+
+import (
+	"errors"
+	"strings"
+)
+
+//FakeSSHCommander is an implementation of provision.SSHCommander to provide predictable responses set by testing code
+//Extend it when needed
+type FakeSSHCommander struct {
+	//Result of the ssh command to look up the FilesystemType
+	FilesystemType string
```

Comment:
Hm, instead of having a field specifically for this, could you maybe make some sort of lookup table field which is `map[string]string`?  Then you could define the result of a given call like so:

```console
cmder := &FakeSSHCommander{
    Responses: map[string]string{
        "stat -f -c %T /var/lib": "btrfs\n",
    },
}
```

WDYT?